### PR TITLE
add metadata to example templates

### LINKED ISSUE #3432: Add resource metadata to example templates
#2989 added the capability to attach metadata to workspace resources and have it displayed in the UI. Demonstrate this functionality by adding metadata to one or more example templates. 

This might involve a new template that exists specifically to show off the metadata feature, but we could also just add useful metadata to some of our existing templates (e.g. IDs of cloud resources such as instances, pods and volumes).

## Definition of Done

We should have at least one workspace template that is usable in its own right and includes metadata. It does not need to comprehensively demonstrate every possible feature (e.g. marking metadata items as "sensitive") if there's no good reason to do so.

### PR DESCRIPTION
**Pull request by bpmct**
resolves #3432

I manually tested each template

### CODE REVIEW

**Review by BrunoQuaresma — APPROVED** [review]
Code LGTM!

PS: Just be careful with the commit message. It is missing the prefix like `chore: My message`

File: examples/templates/aws-linux/main.tf [hunk 1]
```diff
@@ -155,3 +155,19 @@ resource "aws_instance" "dev" {
...
+    key   = "region"
+    value = var.region
+  }
+  item {
+    key   = "instance type"
+    value = aws_instance.dev.instance_type
+  }
+  item {
+    key   = "disk"
+    value = "${aws_instance.dev.root_block_device[0].volume_size} GB"
```
Comment (dwahler) [inline examples/templates/aws-linux/main.tf:171]: Nitpick: I think all of these "volume size" parameters are actually measured in GiB, not decimal GB. How about labeling them that way?

**Review by dwahler — APPROVED** [review]

**Review by dwahler — CHANGES_REQUESTED** [review]

File: examples/templates/docker-code-server/main.tf [hunk 2]
```diff
@@ -81,4 +81,3 @@ resource "docker_container" "workspace" {
     volume_name    = docker_volume.home_volume.name
     read_only      = false
   }
-}
```
[the remark this answers is not in the document]
Comment (bpmct) [reply examples/templates/docker-code-server/main.tf:84]: woahhhh great catch. I guess I didn't test the latest commit manually

File: examples/templates/aws-linux/main.tf (hunk 1 above)
Comment (bpmct) [reply examples/templates/aws-linux/main.tf:171]: for sure!

**Review by dwahler — APPROVED** [review]

**Pull request merged**